src/utils/tests/frame.spec.js

```diff
@@ -118,4 +119,135 @@ describe("function names", () => {
       });
     });
   });
+
+  describe("collapseFrames", () => {
+    it("collapses React component render frames and Webpack bootstrap frames", () => {
+      const frames = [
+        {
+          id: "0",
+          displayName: "render",
+          source: {
+            url: "webpack:///src/App.js"
+          }
+        },
+        {
+          id: "1",
+          displayName:
+            "_renderValidatedComponentWithoutOwnerOrContext/renderedElement<",
+          library: "React",
+          source: {
+            url: "webpack:///~/react-dom/lib/ReactCompositeComponent.js"
+          }
+        },
+        {
+          id: "2",
+          displayName: "render",
+          library: "React",
+          source: {
+            url: "webpack:///~/react-dom/lib/ReactMount.js"
+          }
+        },
+        {
+          id: "3",
+          displayName: "__webpack_require__",
+          library: "Webpack",
+          source: {
+            url: "webpack:///webpack/bootstrap 01d88449ca6e9335a66f"
+          }
+        },
+        {
+          id: "4",
+          displayName: "(global)",
+          source: {
+            url: "https://foobar.com/bundle.js"
+          }
+        }
+      ];
+      const collapsedFrames = collapseFrames(frames);
+
+      expect(collapsedFrames).toEqual([
+        frames[0],
+        [frames[1], frames[2]],
+        [frames[3], frames[4]]
+      ]);
+    });
+  });
+
+  it("groups async frames generated by Babel", () => {
+    const frames = [
+      {
```

Comment:
same here, lots of repeated info, which isnt necessarily crucial for the test. let me know if you need help with this!